crypto/sphinx/src/packet.rs

```diff
@@ -0,0 +1,805 @@
+use hopr_crypto_types::crypto_traits::StreamCipher;
+use hopr_crypto_types::prelude::*;
+use hopr_primitive_types::prelude::*;
+
+use std::fmt::{Debug, Formatter};
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+use typenum::Unsigned;
+
+use crate::errors::SphinxError;
+use crate::routing::{forward_header, SphinxHeaderSpec};
+use crate::surb::{ReplyOpener, SURB};
+use crate::{
+    derivation::derive_packet_tag,
+    routing::{ForwardedHeader, RoutingInfo},
+    shared_keys::{Alpha, GroupElement, SharedKeys, SharedSecret, SphinxSuite},
+};
+
+/// Holds data that are padded up to `P + 1`.
+///
+/// Data in this instance is guaranteed to be always `P + 1` bytes-long.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PaddedPayload<const P: usize>(Box<[u8]>);
+
+impl<const P: usize> PaddedPayload<P> {
+    /// Tag used to separate padding from data
+    pub const PADDING_TAG: u8 = 0xaa;
```

Comment:
Maybe a better solution than a padding tag is to dedicate the first 2 bytes as a length specifier (i.e. data) and the rest will be `MAX_PAYLOAD_SIZE` - `.length` to get the padding data. An advantage in this approach is that on serialization and deserialization this operation is super fast.

Reply:
Nope, see below.